implementation/src/main/java/io/smallrye/graphql/execution/datafetcher/decorator/MetricDecorator.java

```diff
@@ -26,7 +28,8 @@ public void after(DataFetchingEnvironment dfe) {
         if (startTime != null) {
             long duration = System.nanoTime() - startTime;
             MetricID metricID = new MetricID("mp_graphql_" + dfe.getField().getName());
-            MetricRegistries.get(MetricRegistry.Type.VENDOR).simpleTimer(metricID.getName(), metricID.getTagsAsArray())
+            lookupService.getMetricRegistry(MetricRegistry.Type.VENDOR)
```

Comment:
instead of using the lookupService on each request, could we make `MetricsRegistry` a private final field, and then initialize that (using the LookupService) in the constructor?

Reply:
@jmartisk ^^^

Reply:
Good point, done that.